### DISCUSSION TITLE
Extracting peak matrix with peak ranges

### DISCUSSION THREAD

**Discussion by sneddonucsf** [GreenleafLab/ArchR#1392] [Questions / Documentation] [answered]
Hello, 

I have a list of peaks of interest that I have some additional annotation for. I would like to subset my peak set so that I can plot the peaks across pseudotime outside of ArchR with some row annotations (ex. what cell type the peak is associated with). When I get my peak matrix with
```
peaks = getMatrixFromProject(proj, useMatrix = "PeakMatrix")
```
this returns the SE object. However, when I get the peak matrix with
```
PM = assays(peaks)$PeakMatrix
PM[1:3, 1:3]
```
I can see that the rownames of the matrix are not peak ranges
```
3 x 3 sparse Matrix of class "dgCMatrix"
     Arrow1#GAAGTGGAGCCTGTAT-1 Arrow1#ACTCGGGCAAGGCGTA-1 Arrow1#AGGCGTCGTGCACATT-1
[1,]                           .                           .                           .
[2,]                           .                           .                           .
[3,]                           .                           2                           2
```

How can I get the ranges associated with these counts so that I can map my row annotations to the heatmap? Does the matrix rownames follow the same order as the ranges in the PeakSet? Apologies if I am completely misunderstanding this!

**Comment by rcorces** [comment, marked as answer]
https://www.bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html

You're looking for `rowRanges()`

**Reply by sneddonucsf** [reply]
Thank you!
